lib/libzfs/libzfs_mount.c

```diff
@@ -351,14 +351,128 @@ zfs_is_mountable(zfs_handle_t *zhp, char *buf, size_t buflen,
  * in the case of a mount failure we do not have the exact errno.  We must
  * make due with return value from the mount process.
  *
- * In the long term a shared library called libmount is under development
- * which provides a common API to address the locking and errno issues.
- * Once the standard mount utility has been updated to use this library
- * we can add an autoconf check to conditionally use it.
+ * If libmount exists, use libmount API rather than invoking external programs.
+ * libmount provides a common API to address the locking and errno issues.
+ * In modern distros, the system mount utilities are libmount based.
  *
  * http://www.kernel.org/pub/linux/utils/util-linux/libmount-docs/index.html
  */
 
+/*
+ * MNT_EX_* and related API didn't exist until v2.30 released in 2017,
+ * so initially only support distros with libmount version v2.30 or above.
+ */
+#ifdef HAVE_LIBMOUNT
+#include <libmount/libmount.h>
+#endif
+#if (LIBMOUNT_MAJOR_VERSION >= 2) && (LIBMOUNT_MINOR_VERSION >= 30)
```

Comment:
util-linux version has been v2.x for more than a decade, but this should support v3.x, etc. Will fix this.

Might also want to test libmount version in ./configure, so that distros with libmount but has version < 2.30 don't need to link libmount or include libmount.h ?

Reply:
So if major version is 3, does minor version have to be >= 30? if both LIBMOUNT_MAJOR_VERSION and LIBMOUNT_MINOR_VERSION are undefined (due to libmount/libmount.h not included), not sure what happens.

Reply:
It's been v2.x for many years (e.g. v2.5 was tagged in 2006), but if util-linux moves up to 3, the it can't correctly detect things like 3.0, 3.29 and such.

When undefined (header not included), version test results in false, hence fall back on fork/exec.

But the better approach is to just have this version check in `config/user-libmount.m4`. Linking against possibly unused shared library isn't good.

Reply:
> But the better approach is to just have this version check in config/user-libmount.m4. Linking against possibly unused shared library isn't good.

Maybe not. If we're to add more configure-time libmount version checks in future for API compatibility, then we're going to have `HAVE_LIBMOUNT_<version>` variants, and that's just as ugly.

Just in another few years, most people won't be using distros/releases from 2017 or older (i.e. util-linux < 2.30), so in most cases zfs binary will only be linking libmount when libmount symbols are actually used, with exception of distros with conservative policy (namely RHEL7.x in this case).